packages/toast/src/Toast.tsx

```diff
@@ -0,0 +1,335 @@
+  /**
+   * Optional text shown in bold above the body text.
+   */
+  title?: React.ReactNode;
+
+  /**
+   * Required main text for the Toast.
+   */
+  body: React.ReactNode;
+
+  /**
+   * Optional className passed to the wrapping <div /> for the toast.
+   */
+  className?: string;
+
+  /**
+   * Required style variant to render the Toast as.
+   */
+  variant: ToastVariants;
+
+  /**
+   * Optional number between 0 and 1 that sets the progress bar's progress. Note that the progress bar is only rendered when the Toast variant is set to `'progress'`.
+   *
+   * **Default:** `1`
+   */
+  progress?: number;
+
+  /**
+   * Optional boolean that renders the Toast and makes it visible when true.
+   *
+   * **Default:** `false`
+   */
+  open?: boolean;
+
+  /**
+   * Optional click event handler that, when set, renders a close button that receives the passed handler.
+   */
+  close?: React.MouseEventHandler;
+}
+
+export default function Toast({
+  title,
+  body,
+  className,
+  variant,
+  progress = 1,
+  open = false,
+  close,
+  ...rest
+}: ToastProps) {
+  const nodeRef = useRef(null);
+  const dismissible = !!close;
+
+  let VariantIcon: React.ComponentType<any>;
+
+  if (variant === ToastVariants.progress) {
+    VariantIcon = RefreshIcon;
+  } else if (variant === ToastVariants.success) {
+    VariantIcon = CheckmarkWithCircleIcon;
+  } else {
+    VariantIcon = CheckmarkWithCircleIcon;
+  }
+
+  return (
+    <Transition
+      in={open}
+      timeout={transitionDuration}
+      mountOnEnter
+      unmountOnExit
+      nodeRef={nodeRef}
+    >
+      {(state: RTGStates) => (
+        <Portal>
+          <div
+            role="status"
+            ref={nodeRef}
+            className={cx(
+              baseToastStyle,
+              toastVariantStyles[variant],
+              toastTransitionStateStyles[state],
+              className,
+            )}
+            {...rest}
+          >
+            <div
+              className={cx(
+                contentWrapperStyles,
+                contentWrapperVariantStyles[variant],
+                {
+                  [css`
+                    padding-right: calc(${spacing[3]} + ${spacing[2]});
+                  `]: dismissible,
+                },
+              )}
+            >
+              <VariantIcon
+                className={cx(baseIconStyle, iconVariantStyles[variant])}
+                size={30}
+              />
+
+              <div>
+                {title && (
+                  <Body
+                    className={cx(
+                      bodyVariantStyles[variant],
+                      css`
+                        font-weight: bold;
+                      `,
+                    )}
+                  >
+                    {title}
+                  </Body>
+                )}
+
+                <Body className={bodyVariantStyles[variant]}>{body}</Body>
+              </div>
+            </div>
+
+            {dismissible && (
+              <IconButton
```

Comment:
think you mean to add an onClick here to call `close`

Reply:
Whoops! Should be fixed now.